### PR TITLE
Disable failing System.Text.Json tests on Android x86 and on Browser with threading enabled

### PR DESCRIPTION
**Pull request by simonrozsival**
On Android x86, there is a problem with deserialization of doubles.
On Browser with multi-threading enabled, the System.Text.Json tests are timeouting.

This PR disables these tests.

Ref #72101
Ref #65723

/cc @radical

### CODE REVIEW

**Review by krwq — APPROVED** [review]
Are you sure those are the only tests failing? (or most frequent) AFAIK most of our intermittent failures used to happen in random tests and random tests happened to manifest the issue. Approving since this is scoped to Android X86

File: src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs [hunk 1]
```diff
@@ -38,7 +38,8 @@ public static partial class PlatformDetection
         public static bool IsNetBSD => RuntimeInformation.IsOSPlatform(OSPlatform.Create("NETBSD"));
         public static bool IsAndroid => RuntimeInformation.IsOSPlatform(OSPlatform.Create("ANDROID"));
         public static bool IsNotAndroid => !IsAndroid;
-        public static bool IsNotAndroidX86 => !(IsAndroid && IsX86Process);
+        public static bool IsAndroidX86 => IsAndroid && IsX86Process;
+        public static bool IsNotAndroidX86 => !IsAndroidX86;
```
Comment (krwq) [inline src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs:42]: we don't seem to be using this

Comment (simonrozsival) [reply src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs:42]: It's used in a few places to disable some other tests (for example [src/libraries/System.Runtime/tests/System/DoubleTests.cs:1489](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Runtime/tests/System/DoubleTests.cs#L1489))

**Comment by simonrozsival** [comment]
@krwq The tests seem to fail quite consistently on Android x86. I've only disabled the tests that appear in logs referenced in #72101. I'm still looking for more flaky tests that could be related, that's why the PR is still a draft.

File: src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs (hunk 1 above)
Comment (krwq) [reply src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs:42]: ah, I missed it was already there :)

**Review by stephentoub — APPROVED** [review]

**Comment by lewing** [comment]
cc @karelz

**Pull request merged**